keyboards/planck/keymaps/ariccb/keymap.c

```diff
@@ -0,0 +1,675 @@
+  KC_NO,   KC_NO,     KC_NO,      KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, MTLALT_NXT,   KC_NO,   KC_NO,    KC_NO
+),
+
+/* MIT Layout (ADJUST)
+ *
+ * ,-----------------------------------------------------------------------------.
+ * |RGBtog|Ms3 | Ms2 |MsUp | Ms1  |  Hue+|  Hue- | Sat+| Sat- |Brt+ |Brt- | RESET|
+ * |-----------------------------------------------------------------------------|
+ * |RGBMod| MWL | MsL |MDn  |MsR  |GAMING|       |AU_ON|AU_OFF|MU_ON|MU_OF| DEBUG|
+ * |-----------------------------------------------------------------------------|
+ * |     |MWLft|MWUp |NWDn |NWRght|QWERTY|COLEMAK|MI_ON|MI_OF |     |     |MU_Mod|
+ * |-----------------------------------------------------------------------------|
+ * |     |     |     |SLEEP|      |              |     |      |     |     |      |
+ * `-----------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_planck_grid( /* ADJUST LAYER */
+  RGB_TOG, KC_BTN3, KC_BTN2, KC_MS_U, KC_BTN1, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RESET,
+  RGB_MOD, KC_NO,   KC_MS_L, KC_MS_D, KC_MS_R, GAMING,  KC_NO,   AU_ON,   AU_OFF,  MU_ON,   MU_OFF,  DEBUG,
+  KC_TRNS, KC_WH_L, KC_WH_U, KC_WH_D, KC_WH_R, QWERTY,  COLEMAK, MI_ON,   MI_OFF,  KC_TRNS, KC_TRNS, MU_MOD,
+  KC_NO,   KC_NO,   KC_NO,   KC_SLEP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_NO,   KC_NO
+)
+};
+
+#ifdef AUDIO_ENABLE
+float layerswitch_song[][2] = SONG(PLANCK_SOUND);
+float tone_startup[][2]     = SONG(STARTUP_SOUND);
+float tone_qwerty[][2]      = SONG(QWERTY_SOUND);
+float tone_colemak[][2]     = SONG(COLEMAK_SOUND);
+float music_scale[][2]      = SONG(MUSIC_SCALE_SOUND);
+float tone_goodbye[][2]     = SONG(GOODBYE_SOUND);
+
+#endif
+
+bool is_alt_tab_active = false;
+layer_state_t layer_state_set_user(layer_state_t state) {
+
+	static bool is_this_layer_on = false;
+	if (layer_state_cmp(state, 4) != is_this_layer_on) {
+		is_this_layer_on = layer_state_cmp(state, 4);
+		if (is_this_layer_on) {
+			PLAY_SONG(layerswitch_song);
+		}
+		else {
+			stop_all_notes();
+		}
+	}
+  if (is_alt_tab_active) {
+        unregister_code(KC_LALT);
+        is_alt_tab_active = false;
+    }
+    return state;
+
+	switch (get_highest_layer(state)) {
+		case _ADJUST:
+			rgblight_setrgb (0xFF,  0x00, 0x00);
+			break;
+		case _LOWER:
+			rgblight_setrgb (0x00,  0x00, 0xFF);
+			break;
+    case _NUMPAD:
+			rgblight_setrgb (0x00,  0x00, 0xFF);
+			break;
+		case _RAISE:
+			rgblight_setrgb (0x7A,  0x00, 0xFF);
+			break;
+		case _FN:
+			rgblight_setrgb (0x00,  0xFF, 0x00);
+			break;
+		default: //  for any other layers, or the default layer
+			rgblight_setrgb (0xFF,  0xFF, 0xFF);
+			break;
+		}
+	  return state;
+}
+
+void persistant_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```

Comment:
Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer